packages/components/src/components/hds/table/index.ts

```diff
@@ -124,6 +124,8 @@ export default class HdsTable extends Component<HdsTableSignature> {
     if (this.args.identityKey === 'none') {
       return undefined;
     } else {
+      // We return a default key to cause an update operation instead of replacement when the model changes to side-step
+      // a bug in Ember which would cause the table to lose its selection state
```

Comment:
[issue?] I am not sure this comment is correct. We return the `@identity`
https://api.emberjs.com/ember/6.1/classes/Ember.Templates.helpers/methods/each?anchor=each

```suggestion
      // We return `@identity` as default to cause an update operation instead of replacement when the model changes to avoid unexpected side effects
      // see: "Specifying Keys" here: https://api.emberjs.com/ember/6.1/classes/Ember.Templates.helpers/methods/each?anchor=each
      // see also: https://github.com/hashicorp/design-system/pull/1160 + https://github.com/hashicorp/cloud-ui/pull/5178 + https://hashicorp.atlassian.net/browse/HDS-4273
```